### ISSUE TITLE
Add feature to change "date and time added" on selected document records.

### ISSUE THREAD

**Issue by Sohail2949000** [rffrasca/PDFKeeper#32]
**Is your feature request related to a problem? Please describe. **
When we upload Documents the date and time is added at that time but if we upload an old document, we can't change date so it's difficult to search for documents date wise. 
Also, when we delete the serial number for documents does not update it should update so that if we delete any documents, we can have a correct number serial for reference

**Describe the solution you'd like**
It would be very helpful if we change or update date and other information too. and the serial number should rearrange after we delete something

**Comment by rffrasca** [comment]
Hello @Sohail2949000,

Regarding your first request:

I can add a menu item to change the "date added" value. The menu item will show a dialog with a date and time picker control to choose the new date and time.

Are you looking to change one or more documents to the chosen date and time or just the selected document?

Regarding your second request:

The document ID column is auto incrementing so the ID is automatically assigned by the database when a record is inserted into the database to prevent a duplicate ID violation. Reordering the document ID's and updating the built in database sequence is NOT something that I feel should occur every time a document is deleted. If the transaction was to fail and not be able to recover, the database could end up in a state where inserting documents could fail. In addition, some users may reference documents by ID and not want them changing.

**Comment by Sohail2949000** [comment]
Hello Sir
thank you for your quick reply

In my case I need it for selected Items only because I started using PDFKeeper  in 2023, but I have invoices and other documents starting from 2019 so If I can change date to same as document it very helpful to search and find in future.

Also Sir the title can't be changed once uploaded. 
I uploaded an invoice and after uploading the customer request to cancel the invoice, so I had to upload same document but with other title in order to show it's cancelled. If we can update the title it's easy to update changes.


Is there any limit on documents, how many we can upload or what size should we upload so that after all documents it doesn't slowdown. (I am using SQLite single user)

**Comment by rffrasca** [comment]
I can add the changing of "date added" on selected documents into the next version. When I have more information, I will update this issue.

Regarding the changing of the Title. I added that capability along with the changing of the author and subject on selected documents in v9.1.0, released last December.

Regarding the limit on documents. SQLite does have a size limit but, it shouldn't be a concern.
https://www.sqlite.org/limits.html.

To help with database size management and performance, I have added an option to PDFKeeper (to be included in the next version) that when enabled will trigger an SQLite Vacuum on the database after a delete operation to remove all deleted records from the DOCS table and virtual table used for Full-Text Search. SQLite does have an auto vacuum, but it doesn't handle virtual tables and was not implemented in PDFKeeper. So, calling vacuum after each delete operation is the most effective method of ensuring no deleted traces remain in the database.
